### DISCUSSION TITLE
WS2812 support

### DISCUSSION THREAD

**Discussion by haaxema** [HASwitchPlate/openHASP#263] [Ideas]
Hi, 
I am getting started with openHasp and I am very pleased with the way it works.

There is one thing I would like to be added, and that is ws2812(b) support for moodlight leds.
It saves a lot on wiring and extra components when you want to use multiple LED's as moodlight.
(an ESP32 can sink up to 28mA per port, and RGB LEDS user between 10-20 mA per colour.......)

Thanks in advance

Martin

**Comment by fvanroie** [comment]
I have been pondering SK6812 support. Just didn't dig into it yet... What would be a good (small/flexible) library that supports both SK6812 and WS2812?

**Reply by haaxema** [reply]
Take A look at https://blog.ja-ke.tech/2019/06/02/neopixel-performance.html

It seems that neopixel is the smallest.

**Reply by haaxema** [reply]
@fvanroie, are there any updates on this topic?

**Comment by mattkerrison** [comment]
This would be great, I was hoping to add a WS2812B to WT32 for a moodlight as it'd be much easier to wire. so much support for them with things like the wled project

**Reply by DSRocker** [reply]
I would like to use WS2812, too...
How to wire the LEDs and how to configure it?

**Comment by fake-name** [comment]
This would be super handy. I've been playing with a m5stack AWS development kit, which has 10 built-in SK6812 LEDs. 

It'd be nice to be able to do stuff with them.

**Comment by haaxema** [comment]
@fvanroie,  are there any updates on this topic?

**Reply by fvanroie** [reply]
There is no update on this topic...

**Comment by brdcg** [comment]
Pushing this up a bit.
WS2812 Support would be very appreciated!
